src/transformers/modeling_utils.py

```diff
@@ -1160,7 +1160,7 @@ def num_parameters(self, only_trainable: bool = False, exclude_embeddings: bool
                 # used for the 4bit quantization (uint8 tensors are stored)
                 if is_loaded_in_4bit and isinstance(param, bnb.nn.Params4bit):
                     total_numel.append(
-                        param.numel() * 2 * self.hf_quantizer.quantization_config.bnb_4bit_quant_storage.itemsize
+                        param.numel() * 2 * self.hf_quantizer.quantization_config.bnb_4bit_quant_storage.element_size()
```

Comment:
I meant to respond to the other one earlier, but may be good to have an if/else to know which version to call? E.g. `param.numel() * 2 * elem_size` where `elemsize` is defined earlier based on the torch version

Reply:
Correction, as @BenjaminBossan pointed out this is a must for us to have

Reply:
Besides itemsize not being available in torch 2.0.1, I couldn't get element_size() to work either (not a dtype attribute error).   

I put this 'helper' function at the top of the file to do it manually:

def get_dtype_size(dtype):
    if dtype == torch.float32:
        return 4
    elif dtype == torch.float64:
        return 8
    elif dtype == torch.float16:
        return 2
    elif dtype == torch.uint8:
        return 1
    elif dtype == torch.int8:
        return 1
    elif dtype == torch.int16:
        return 2
    elif dtype == torch.int32:
        return 4
    elif dtype == torch.int64:
        return 8
    else:
        raise ValueError("Unsupported dtype")

and then used this in place of the itemsize code:

if is_loaded_in_4bit and isinstance(param, bnb.nn.Params4bit):
    quant_storage = self.hf_quantizer.quantization_config.bnb_4bit_quant_storage
    nb_params = get_dtype_size(quant_storage)
    total_numel.append(param.numel() * 2 * nb_params)
else:
    total_numel.append(param.numel())
    
I've only tested it by using the qlora config on tiny-llama; and at least for this case it works:

   accelerate launch -m axolotl.cli.train examples/tiny-llama/qlora.yml

I am using image 'winglian/axolotl-base:main-base-py3.11-cu121-2.2.1' (which actually uses torch 2.0.1 not 2.2.1 as the name implies.).  Ubuntu 22.04 / laptop RTX4090 (16G)

Reply:
Thank you soooo much!!!